activesupport/test/number_helper_test.rb

```diff
@@ -344,6 +344,9 @@ def test_number_to_human_with_custom_units
           #Spaces are stripped from the resulting string
           assert_equal "4", number_helper.number_to_human(4, units: { unit: "", ten: "tens " })
           assert_equal "4.5  tens", number_helper.number_to_human(45, units: { unit: "", ten: " tens   " })
+
+          #Uses only the provided units and does not try to use larger ones
+          assert_equal "1000 kilometers", number_helper.number_to_human(1_000_000, units: {unit: "meter", thousand: "kilometers"})
```

Comment:
May I suggest you also add these examples (in the “gangster” group, right after line 341):
```rb
assert_equal "1000 hundred bucks", number_helper.number_to_human(100_000, :units => gangster)
assert_equal "1 thousand quids", number_helper.number_to_human(999_999, :units => gangster)
assert_equal "1 thousand quids", number_helper.number_to_human(1_000_000, :units => gangster)
```
The first 2 fail on master, exposing another bug introduced by #20872. All 3 pass using your code.